pkg/controller/operconfig/ingressconfig_controller.go

```diff
@@ -80,17 +65,13 @@ func (r *ReconcileIngressConfigs) Reconcile(request reconcile.Request) (reconcil
 }
 
 // setLabelsOnNamespace sets the labels specified on the target namespace using the client API
-func (r *ReconcileIngressConfigs) setLabelsOnNamespace(targetNamespace string, labels map[string]string) error {
+func (r *ReconcileIngressConfigs) setLabelsOnNamespace(targetNamespace string, labels map[string]*string) error {
 	var err error
 	namespace := &corev1.Namespace{TypeMeta: metav1.TypeMeta{APIVersion: corev1.SchemeGroupVersion.String(), Kind: "Namespace"}}
 	err = r.client.Get(context.TODO(), types.NamespacedName{Name: targetNamespace}, namespace)
 	if err != nil {
 		if apierrors.IsNotFound(err) {
-			log.Printf("Namespace %s was deleted", targetNamespace)
-			// Request object not found, could have been deleted after reconcile request.
-			// Owned objects are automatically garbage collected, since we set
-			// the ownerReference (see https://kubernetes.io/docs/concepts/workloads/controllers/garbage-collection/).
-			// Return and don't requeue
+			log.Printf("Namespace %s was deleted or doesn't exist", targetNamespace)
```

Comment:
There are two possibilities here:
1. if we expected `openshift-host-namespace` to exist, and it doesn't, then that's an error
2. if we didn't expect `openshift-host-namespace` to exist, then we shouldn't even be here.

CNO _should not_ log a message about `openshift-host-namespace` not existing once every five minutes in a cluster where we didn't expect it to exist in the first place.

Reply:
Makes sense. Ideally we don't even want to create a ingresscontrollerconfig reconciler if the plugin is not openshift-sdn or ovn-kubernetes. But I think at operconfig_controller creation time, we don't know. The alternative is to detect the plugin during each reconcile loop. I think that probably makes the most sense. Thoughts?